main.go

```diff
@@ -53,13 +56,34 @@ func reverseProxy(rw http.ResponseWriter, req *http.Request) {
 	var upstreamUrl *url.URL
 	var enforceFunc func(string, map[string]bool) (string, error)
 	var tenantLabels map[string]bool
-	query := req.URL.Query().Get("query")
+	var err error
+
+	urlKey := "query"
+	if containsApiV1Labels(req.URL.Path) {
```

Comment:
hier wieder dasselbe wie oben für die Query und die Enforcerfunction. 
Wäre es nicht schöner, den request einfach ganz am Anfang mal anzunehmen, und dann zu verarbeiten und entsprechend weiterzuleiten, anstatt alles hier zu machen?

Reply:
Alternativ 2 Endpoints anbieten am proxy für /logs und /metrics. Die Datasources sind in Grafana sowieso einzeln eingebunden. Dann sparst du dir viele unschöne if Abfragen.